src/pkg/notifier/formats/formatter.go

```diff
@@ -0,0 +1,50 @@
+//  Copyright Project Harbor Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package formats
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/goharbor/harbor/src/lib/errors"
+	"github.com/goharbor/harbor/src/pkg/notifier/model"
+)
+
+var (
+	// defaultFormatter is the global single formatter for Default.
+	defaultFormatter Formatter = &Default{}
+	// cloudEventsFormatter is the global single formatter for CloudEvents.
+	cloudEventsFormatter Formatter = &CloudEvents{}
+)
+
+// Formatter is the interface for event which for implementing different drivers to
+// organize their customize data format.
+type Formatter interface {
+	// Format formats the data to expected format and return request headers and encoded payload
+	Format(context.Context, *model.HookEvent) (http.Header, []byte, error)
+}
+
+// GetFormatter returns corresponding formatter from format type.
+func GetFormatter(formatType string) (Formatter, error) {
```

Comment:
Prefer map[]Formatter instead of switch case

Reply:
updated.